src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -0,0 +1,706 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Globalization;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    // Portions of the code implemented below are based on the 'Berkeley SoftFloat Release 3e' algorithms.
+
+    /// <summary>
+    /// An IEEE 754 compliant float16 type.
+    /// </summary>
+    [Serializable]
+    [StructLayout(LayoutKind.Sequential)]
+    public readonly struct Half : IComparable, IFormattable, IComparable<Half>, IEquatable<Half>, ISpanFormattable
+    {
+        private const NumberStyles DefaultParseStyle = NumberStyles.Float | NumberStyles.AllowThousands;
+
+        // Constants for manipulating the private bit-representation
+
+        private const ushort SignMask = 0x8000;
+        private const ushort SignShift = 15;
+
+        private const ushort ExponentMask = 0x7C00;
+        private const ushort ExponentShift = 10;
+
+        private const ushort SignificandMask = 0x03FF;
+        private const ushort SignificandShift = 0;
+
+        private const ushort MinSign = 0;
+        private const ushort MaxSign = 1;
+
+        private const ushort MinExponent = 0x00;
+        private const ushort MaxExponent = 0x1F;
+
+        private const ushort MinSignificand = 0x0000;
+        private const ushort MaxSignificand = 0x03FF;
+
+        // Constants representing the private bit-representation for various default values
+
+        private const ushort PositiveZeroBits = 0x0000;
+        private const ushort NegativeZeroBits = 0x8000;
+
+        private const ushort EpsilonBits = 0x0001;
+
+        private const ushort PositiveInfinityBits = 0x7C00;
+        private const ushort NegativeInfinityBits = 0xFC00;
+
+        private const ushort PositiveQNaNBits = 0x7E00;
+        private const ushort NegativeQNaNBits = 0xFE00;
+
+        private const ushort MinValueBits = 0xFBFF;
+        private const ushort MaxValueBits = 0x7BFF;
+
+
+        //
+        // Well-defined and commonly used values
+        //
+
+        public static readonly Half Epsilon = new Half(EpsilonBits);                        //  5.9604645E-08
```

Comment:
Is it intentional for these to be readonly fields and not properties?

Fields are harder to optimize than properties. I think these should be properties. It is what we have standardized on in other places, e.g. `IntPtr.MaxValue` recently.

Reply:
This was likely just oversight due to the proposal being 3 years old. I will raise it at the start of tomorrows meeting but don't expect any pushback on these being properties.

CC. @terrajobst , @dotnet/fxdc

Reply:
Made them properties. If there's pushback tomorrow, I'll make them fields again.

Reply:
This got covered at the very beginning of API review, no pushback.